update.py

```diff
@@ -219,6 +226,65 @@ def copy_build(site):
         shutil.rmtree(olddir)
 
 
+def copy_and_keep_build(site):
+    """
+    Copies each site into the target location and keep last "n" builds as backups
+    """
+    global error_count
+    for wiki in ALL_WIKIS:
+        if site=='common':
+            continue
+        if site is not None and site != wiki:
+            continue
+        debug('copy: %s' % wiki)
+        targetdir = os.path.join(args.destdir, wiki)
+
+        debug("Creating temporary folders")
+        previousdir = os.path.join(args.backupdestdir)
+        debug('Recreating %s' % previousdir )
+        if not os.path.exists(previousdir):
+            os.mkdir(previousdir)
+    
+        olddir = os.path.join(previousdir, str(building_time) )
+        debug('Recreating %s' % olddir )
+        if not os.path.exists(olddir):
+            os.mkdir(olddir)
+
+        if os.path.exists(targetdir):
+            debug('Moving %s into %s' % (targetdir,olddir) )
+            shutil.move(targetdir, olddir)
+
+        sourcedir='./%s/build/html/' % wiki
+
+        html_moved_dir = os.path.join(args.destdir, 'html')
+        try:
+            subprocess.check_call(['mv', sourcedir, html_moved_dir])
+            # Rename move! (single move to html/* failed)
+            subprocess.check_call(['mv', html_moved_dir ,targetdir])
+            debug("Moved to %s" % targetdir)
+        except:
+            error("FAIL moving output to %s" % targetdir)
+        finally:
+            debug("Creating a backup in %s" % olddir)
+            subprocess.check_call(['cp', '-r', targetdir ,olddir])
+            delete_old_wiki_backups(previousdir, N_BACKUPS_RETAIN)
+
+
+def delete_old_wiki_backups(folder, n_to_keep):
+    try:
+        debug('Checking number of number of backups in folder %s' % folder)
+        backup_folders = glob.glob(folder  + "/*/")
+        backup_folders.sort()
+        if len(backup_folders) > n_to_keep:
+            for i in range(0, len(backup_folders) - n_to_keep):
+                debug('Deleting folder %s' % str(backup_folders[i]))
+                shutil.rmtree(str(backup_folders[i]))
+        else:
+            debug('No old backups to delete in %s' % folder)
+    except Exception as e:
+        error('Error on deleting some previous wiki backup folders: %s' % e)
```

Comment:
Should we be incrementing `error_count` here?

Similarly in some other exception blocks, actually.

Reply:
This, specifically, is a good sign that directories that have not been deleted could fill the disk.

I took as a standard accounting in all, just in case.

Anyway, given this review, I see that I will have to go back with some previous PR to adjust some things. Maybe evaluate everything concerning python 3 and also with the same libraries.